src/analysis.rs

```diff
@@ -1,26 +1,235 @@
+use sketches_ddsketch::{Config, DDSketch};
+
 use std::{
     collections::{hash_map::RandomState, BTreeSet, HashMap, HashSet},
     hash::{BuildHasher, Hasher},
     io::Write,
 };
 
-use histo::Histogram;
+use thiserror::Error;
+use lading_payload::dogstatsd::{KindWeights, MetricWeights};
 
 use crate::{
     dogstatsdmsg::{DogStatsDMetricType, DogStatsDMsg, DogStatsDMsgKind},
     dogstatsdreader::DogStatsDReader,
 };
 
-const DEFAULT_NUM_BUCKETS: u64 = 10;
+type KindCount = (u32, Option<HashMap<DogStatsDMetricType, u32>>);
```

Comment:
Ah, good to know, thanks!
I'll leave this as-is -- feels more semantically correct since `Event` and `ServiceChecks` don't need the metric counter map.